runtime/domain.c

```diff
@@ -1520,10 +1532,16 @@ void caml_interrupt_self(void) {
 
 void caml_reset_young_limit(caml_domain_state * dom_st)
 {
-  CAMLassert ((uintnat)dom_st->young_ptr > (uintnat)dom_st->young_trigger);
+  /* trigger at the greater of young_trigger and memprof_young_trigger */
+  value *trigger = dom_st->young_trigger;
+  if (dom_st->memprof_young_trigger &&
+      (dom_st->memprof_young_trigger > trigger)) {
```

Comment:
If I remember well, in the previous implementation, the default value for `memprof_young_trigger` was `young_alloc_start` when memprof was not currently running.

It seems like you are in a somewhat redundant situation, where you both use `NULL` and `young_alloc_start`. Do I understand well? Any reason for this? I'm 100% fine with any situation (if I remember well, the choice of `young_alloc_start` was guided by the fact that the implementation was slightly simpler in this case), but it is weird to use both conventions.

Reply:
(In any case, please document and add an assertion checking the possible values for `memprof_young_trigger`.)

Reply:
Good catch. This was left-over from an earlier version of my multicore statmemprof code, in which memprof_young_trigger was only set if there was a memprof profile running. I'll simplify this away, and add an assert.